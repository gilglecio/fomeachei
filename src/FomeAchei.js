/*
 * @flow
 */
import React, { Component } from 'react';
import { Scene, Router } from 'react-native-router-flux';
import moment from 'moment';

import 'moment/locale/pt-br';

import Main from './containers/Main';

import Dash from './components/Dash';
import Found from './components/Found';
import FoundIt from './components/FoundIt';
import Hungry from './components/Hungry';

moment.locale('pt-br');

/**
 * FomeAchei main component
 *
 * The best experience to colaborative information about found foods
 *
 */
export default class FomeAchei extends Component {

  /**
   * Render the React Native Router Flux schema
   * @method render
   * @return {Router} Router Schema
   */
  render() {
    return (
      <Router>
        <Scene key="root">
          <Scene key="main" component={Main} initial hideNavBar>
            <Scene key="dash" component={Dash} hideNavBar initial />
            <Scene key="found" component={Found} hideNavBar />
            <Scene key="foundIt" component={FoundIt} hideNavBar />
            <Scene key="hungry" component={Hungry} hideNavBar />
          </Scene>
        </Scene>
      </Router>
    );
  }
}
