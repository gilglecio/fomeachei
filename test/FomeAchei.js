import 'react-native';
import React from 'react';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

import FomeAchei from '../src/FomeAchei';

it('renders correctly', () => {
  const tree = renderer.create(
    <FomeAchei />
  );

  expect(tree).toMatchSnapshot();
});
