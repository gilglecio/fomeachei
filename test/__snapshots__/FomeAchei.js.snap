exports[`test renders correctly 1`] = `
<View
  onLayout={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
        "flex": 1,
      },
      undefined,
    ]
  }>
  <View
    style={
      Object {
        "flex": 1,
      }
    }>
    <View
      onMoveShouldSetResponder={[Function]}
      onMoveShouldSetResponderCapture={[Function]}
      onResponderEnd={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderReject={[Function]}
      onResponderRelease={[Function]}
      onResponderStart={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      onStartShouldSetResponderCapture={[Function]}
      pointerEvents="auto"
      style={
        Object {
          "backgroundColor": "white",
          "bottom": 0,
          "left": 0,
          "opacity": 1,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "transform": Array [
            Object {
              "translateX": 0,
            },
            Object {
              "translateY": 0,
            },
          ],
        }
      }>
      <View
        style={
          Array [
            Object {
              "flex": 1,
            },
            undefined,
          ]
        }>
        <View
          onLayout={[Function]}
          style={
            Object {
              "bottom": 0,
              "justifyContent": "center",
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }>
          <View
            style={
              Array [
                Object {
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                },
                Object {
                  "right": 250,
                },
              ]
            }>
            <ScrollView
              scrollsToTop={false}
              style={
                Object {
                  "backgroundColor": "#272B34",
                  "borderBottomColor": "transparent",
                  "borderColor": "transparent",
                  "flex": 1,
                }
              }>
              <View
                style={
                  Object {
                    "backgroundColor": "#272B34",
                    "borderBottomColor": "transparent",
                    "borderColor": "transparent",
                    "flex": 1,
                  }
                }>
                <View
                  style={
                    Array [
                      Object {
                        "backgroundColor": "#ffffff",
                        "borderBottomWidth": 1,
                        "borderColor": "#cbd2d9",
                        "borderTopWidth": 1,
                        "marginTop": 20,
                      },
                      Object {
                        "backgroundColor": "#272B34",
                        "borderBottomColor": "transparent",
                        "borderColor": "transparent",
                        "flex": 1,
                      },
                    ]
                  }>
                  <View
                    accessibilityComponentType={undefined}
                    accessibilityLabel={undefined}
                    accessibilityTraits={undefined}
                    accessible={true}
                    hitSlop={undefined}
                    onLayout={undefined}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      Object {
                        "opacity": 1,
                      }
                    }
                    testID={undefined}>
                    <View
                      onLongPress={undefined}
                      onPress={undefined}
                      style={
                        Array [
                          Object {
                            "backgroundColor": "transparent",
                            "borderBottomColor": "#ededed",
                            "borderBottomWidth": 1,
                            "paddingBottom": 10,
                            "paddingRight": 10,
                            "paddingTop": 10,
                          },
                          Object {
                            "backgroundColor": "#272B34",
                            "borderBottomColor": "transparent",
                            "borderColor": "transparent",
                            "flex": 1,
                          },
                        ]
                      }
                      underlayColor="white">
                      <View
                        style={
                          Array [
                            Object {
                              "flexDirection": "row",
                              "marginLeft": 10,
                            },
                            undefined,
                          ]
                        }>
                        <View
                          onLongPress={undefined}
                          onPress={undefined}
                          style={
                            Array [
                              false,
                              false,
                              false,
                              Object {
                                "alignItems": "center",
                                "backgroundColor": "transparent",
                                "justifyContent": "center",
                              },
                              undefined,
                            ]
                          }
                          underlayColor="white">
                          <Text
                            accessible={true}
                            allowFontScaling={false}
                            ellipsizeMode="tail"
                            style={
                              Array [
                                Object {
                                  "color": "#E2E3E5",
                                  "fontSize": 24,
                                },
                                Array [
                                  Object {
                                    "backgroundColor": "transparent",
                                  },
                                  Array [
                                    Object {
                                      "marginRight": 8,
                                    },
                                    undefined,
                                  ],
                                ],
                                Object {
                                  "fontFamily": "Material Icons",
                                  "fontStyle": "normal",
                                  "fontWeight": "normal",
                                },
                              ]
                            }>
                            î¢¶
                          </Text>
                        </View>
                        <View
                          style={
                            Object {
                              "flex": 1,
                              "justifyContent": "center",
                            }
                          }>
                          <Text
                            accessible={true}
                            allowFontScaling={true}
                            ellipsizeMode="tail"
                            style={
                              Array [
                                Object {
                                  "fontFamily": undefined,
                                },
                                undefined,
                                undefined,
                                undefined,
                                undefined,
                                undefined,
                                undefined,
                                undefined,
                                undefined,
                                undefined,
                                Array [
                                  Object {
                                    "color": "#43484d",
                                    "fontSize": 17.5,
                                  },
                                  Object {
                                    "color": "#E2E3E5",
                                    "fontSize": 18,
                                    "marginLeft": 14,
                                  },
                                  false,
                                  undefined,
                                ],
                              ]
                            }>
                            Buscar Alimento
                          </Text>
                          <View
                            style={undefined} />
                        </View>
                        <View
                          style={
                            Object {
                              "alignItems": "flex-end",
                              "flex": 0.15,
                              "justifyContent": "center",
                            }
                          }>
                          <View
                            onLongPress={undefined}
                            onPress={undefined}
                            style={
                              Array [
                                false,
                                false,
                                false,
                                Object {
                                  "alignItems": "center",
                                  "backgroundColor": "transparent",
                                  "justifyContent": "center",
                                },
                                undefined,
                              ]
                            }
                            underlayColor="white">
                            <Text
                              accessible={true}
                              allowFontScaling={false}
                              ellipsizeMode="tail"
                              style={
                                Array [
                                  Object {
                                    "color": "#bdc6cf",
                                    "fontSize": 28,
                                  },
                                  Array [
                                    Object {
                                      "backgroundColor": "transparent",
                                    },
                                    undefined,
                                  ],
                                  Object {
                                    "fontFamily": "Material Icons",
                                    "fontStyle": "normal",
                                    "fontWeight": "normal",
                                  },
                                ]
                              }>
                              î
                            </Text>
                          </View>
                        </View>
                      </View>
                    </View>
                  </View>
                  <View
                    accessibilityComponentType={undefined}
                    accessibilityLabel={undefined}
                    accessibilityTraits={undefined}
                    accessible={true}
                    hitSlop={undefined}
                    onLayout={undefined}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      Object {
                        "opacity": 1,
                      }
                    }
                    testID={undefined}>
                    <View
                      onLongPress={undefined}
                      onPress={undefined}
                      style={
                        Array [
                          Object {
                            "backgroundColor": "transparent",
                            "borderBottomColor": "#ededed",
                            "borderBottomWidth": 1,
                            "paddingBottom": 10,
                            "paddingRight": 10,
                            "paddingTop": 10,
                          },
                          Object {
                            "backgroundColor": "#272B34",
                            "borderBottomColor": "transparent",
                            "borderColor": "transparent",
                            "flex": 1,
                          },
                        ]
                      }
                      underlayColor="white">
                      <View
                        style={
                          Array [
                            Object {
                              "flexDirection": "row",
                              "marginLeft": 10,
                            },
                            undefined,
                          ]
                        }>
                        <View
                          onLongPress={undefined}
                          onPress={undefined}
                          style={
                            Array [
                              false,
                              false,
                              false,
                              Object {
                                "alignItems": "center",
                                "backgroundColor": "transparent",
                                "justifyContent": "center",
                              },
                              undefined,
                            ]
                          }
                          underlayColor="white">
                          <Text
                            accessible={true}
                            allowFontScaling={false}
                            ellipsizeMode="tail"
                            style={
                              Array [
                                Object {
                                  "color": "#E2E3E5",
                                  "fontSize": 24,
                                },
                                Array [
                                  Object {
                                    "backgroundColor": "transparent",
                                  },
                                  Array [
                                    Object {
                                      "marginRight": 8,
                                    },
                                    undefined,
                                  ],
                                ],
                                Object {
                                  "fontFamily": "Material Icons",
                                  "fontStyle": "normal",
                                  "fontWeight": "normal",
                                },
                              ]
                            }>
                            î§
                          </Text>
                        </View>
                        <View
                          style={
                            Object {
                              "flex": 1,
                              "justifyContent": "center",
                            }
                          }>
                          <Text
                            accessible={true}
                            allowFontScaling={true}
                            ellipsizeMode="tail"
                            style={
                              Array [
                                Object {
                                  "fontFamily": undefined,
                                },
                                undefined,
                                undefined,
                                undefined,
                                undefined,
                                undefined,
                                undefined,
                                undefined,
                                undefined,
                                undefined,
                                Array [
                                  Object {
                                    "color": "#43484d",
                                    "fontSize": 17.5,
                                  },
                                  Object {
                                    "color": "#E2E3E5",
                                    "fontSize": 18,
                                    "marginLeft": 14,
                                  },
                                  false,
                                  undefined,
                                ],
                              ]
                            }>
                            Mapear Alimento
                          </Text>
                          <View
                            style={undefined} />
                        </View>
                        <View
                          style={
                            Object {
                              "alignItems": "flex-end",
                              "flex": 0.15,
                              "justifyContent": "center",
                            }
                          }>
                          <View
                            onLongPress={undefined}
                            onPress={undefined}
                            style={
                              Array [
                                false,
                                false,
                                false,
                                Object {
                                  "alignItems": "center",
                                  "backgroundColor": "transparent",
                                  "justifyContent": "center",
                                },
                                undefined,
                              ]
                            }
                            underlayColor="white">
                            <Text
                              accessible={true}
                              allowFontScaling={false}
                              ellipsizeMode="tail"
                              style={
                                Array [
                                  Object {
                                    "color": "#bdc6cf",
                                    "fontSize": 28,
                                  },
                                  Array [
                                    Object {
                                      "backgroundColor": "transparent",
                                    },
                                    undefined,
                                  ],
                                  Object {
                                    "fontFamily": "Material Icons",
                                    "fontStyle": "normal",
                                    "fontWeight": "normal",
                                  },
                                ]
                              }>
                              î
                            </Text>
                          </View>
                        </View>
                      </View>
                    </View>
                  </View>
                  <View
                    accessibilityComponentType={undefined}
                    accessibilityLabel={undefined}
                    accessibilityTraits={undefined}
                    accessible={true}
                    hitSlop={undefined}
                    onLayout={undefined}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      Object {
                        "opacity": 1,
                      }
                    }
                    testID={undefined}>
                    <View
                      onLongPress={undefined}
                      onPress={undefined}
                      style={
                        Array [
                          Object {
                            "backgroundColor": "transparent",
                            "borderBottomColor": "#ededed",
                            "borderBottomWidth": 1,
                            "paddingBottom": 10,
                            "paddingRight": 10,
                            "paddingTop": 10,
                          },
                          Object {
                            "backgroundColor": "#272B34",
                            "borderBottomColor": "transparent",
                            "borderColor": "transparent",
                            "flex": 1,
                          },
                        ]
                      }
                      underlayColor="white">
                      <View
                        style={
                          Array [
                            Object {
                              "flexDirection": "row",
                              "marginLeft": 10,
                            },
                            undefined,
                          ]
                        }>
                        <View
                          onLongPress={undefined}
                          onPress={undefined}
                          style={
                            Array [
                              false,
                              false,
                              false,
                              Object {
                                "alignItems": "center",
                                "backgroundColor": "transparent",
                                "justifyContent": "center",
                              },
                              undefined,
                            ]
                          }
                          underlayColor="white">
                          <Text
                            accessible={true}
                            allowFontScaling={false}
                            ellipsizeMode="tail"
                            style={
                              Array [
                                Object {
                                  "color": "#E2E3E5",
                                  "fontSize": 24,
                                },
                                Array [
                                  Object {
                                    "backgroundColor": "transparent",
                                  },
                                  Array [
                                    Object {
                                      "marginRight": 8,
                                    },
                                    undefined,
                                  ],
                                ],
                                Object {
                                  "fontFamily": "Material Icons",
                                  "fontStyle": "normal",
                                  "fontWeight": "normal",
                                },
                              ]
                            }>
                            î­
                          </Text>
                        </View>
                        <View
                          style={
                            Object {
                              "flex": 1,
                              "justifyContent": "center",
                            }
                          }>
                          <Text
                            accessible={true}
                            allowFontScaling={true}
                            ellipsizeMode="tail"
                            style={
                              Array [
                                Object {
                                  "fontFamily": undefined,
                                },
                                undefined,
                                undefined,
                                undefined,
                                undefined,
                                undefined,
                                undefined,
                                undefined,
                                undefined,
                                undefined,
                                Array [
                                  Object {
                                    "color": "#43484d",
                                    "fontSize": 17.5,
                                  },
                                  Object {
                                    "color": "#E2E3E5",
                                    "fontSize": 18,
                                    "marginLeft": 14,
                                  },
                                  false,
                                  undefined,
                                ],
                              ]
                            }>
                            Mapeados por mim
                          </Text>
                          <View
                            style={undefined} />
                        </View>
                        <View
                          style={
                            Object {
                              "alignItems": "flex-end",
                              "flex": 0.15,
                              "justifyContent": "center",
                            }
                          }>
                          <View
                            onLongPress={undefined}
                            onPress={undefined}
                            style={
                              Array [
                                false,
                                false,
                                false,
                                Object {
                                  "alignItems": "center",
                                  "backgroundColor": "transparent",
                                  "justifyContent": "center",
                                },
                                undefined,
                              ]
                            }
                            underlayColor="white">
                            <Text
                              accessible={true}
                              allowFontScaling={false}
                              ellipsizeMode="tail"
                              style={
                                Array [
                                  Object {
                                    "color": "#bdc6cf",
                                    "fontSize": 28,
                                  },
                                  Array [
                                    Object {
                                      "backgroundColor": "transparent",
                                    },
                                    undefined,
                                  ],
                                  Object {
                                    "fontFamily": "Material Icons",
                                    "fontStyle": "normal",
                                    "fontWeight": "normal",
                                  },
                                ]
                              }>
                              î
                            </Text>
                          </View>
                        </View>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </ScrollView>
          </View>
          <View
            onMoveShouldSetResponder={[Function]}
            onMoveShouldSetResponderCapture={[Function]}
            onResponderEnd={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderReject={[Function]}
            onResponderRelease={[Function]}
            onResponderStart={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            onStartShouldSetResponderCapture={[Function]}
            style={
              Object {
                "backgroundColor": "transparent",
                "flex": 1,
                "height": 1334,
                "left": 0,
                "position": "absolute",
                "top": 0,
                "transform": Array [
                  Object {
                    "translateX": 0,
                  },
                ],
                "width": 750,
              }
            }>
            <View
              style={
                Object {
                  "backgroundColor": "#373849",
                  "flexDirection": "row",
                  "height": 60,
                }
              }>
              <View
                accessibilityComponentType={undefined}
                accessibilityLabel={undefined}
                accessibilityTraits={undefined}
                accessible={true}
                hitSlop={undefined}
                onLayout={undefined}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                    "padding": 15,
                    "width": 70,
                  }
                }
                testID={undefined}>
                <Text
                  accessible={true}
                  allowFontScaling={false}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {
                        "color": "#FFF",
                        "fontSize": 40,
                      },
                      undefined,
                      Object {
                        "fontFamily": "Material Icons",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                    ]
                  }>
                  î
                </Text>
              </View>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "#E2E3E5",
                    "flex": 2,
                    "fontSize": 22,
                    "letterSpacing": 1,
                    "lineHeight": 52,
                    "marginTop": 8,
                    "textAlign": "left",
                  }
                }>
                Fome? Achei!
              </Text>
            </View>
            <View
              style={
                Array [
                  Object {
                    "flex": 1,
                  },
                  undefined,
                ]
              }>
              <View
                style={
                  Array [
                    Object {
                      "backgroundColor": "#373849",
                      "flex": 1,
                      "paddingBottom": 20,
                      "paddingLeft": 10,
                      "paddingRight": 10,
                      "paddingTop": 20,
                    },
                    Object {
                      "alignItems": "center",
                      "justifyContent": "center",
                    },
                  ]
                }>
                <View
                  accessibilityComponentType={undefined}
                  accessibilityLabel={undefined}
                  accessibilityTraits={undefined}
                  accessible={true}
                  hitSlop={undefined}
                  onLayout={undefined}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "alignItems": "center",
                      "borderColor": "#FA5D43",
                      "borderRadius": 100,
                      "borderWidth": 6,
                      "height": 200,
                      "justifyContent": "center",
                      "marginBottom": 20,
                      "marginLeft": 0,
                      "marginRight": 0,
                      "opacity": 1,
                      "width": 200,
                    }
                  }
                  testID={undefined}>
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                      Array [
                        Object {
                          "fontSize": 46,
                        },
                        Object {
                          "color": "#FA5D43",
                        },
                      ]
                    }>
                    Fome?
                  </Text>
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                      Array [
                        Object {
                          "fontSize": 16,
                        },
                        Object {
                          "color": "#FA5D43",
                        },
                      ]
                    }>
                    Encontrar Alimento
                  </Text>
                </View>
                <View
                  accessibilityComponentType={undefined}
                  accessibilityLabel={undefined}
                  accessibilityTraits={undefined}
                  accessible={true}
                  hitSlop={undefined}
                  onLayout={undefined}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "alignItems": "center",
                      "borderColor": "#4DBA75",
                      "borderRadius": 100,
                      "borderWidth": 6,
                      "height": 200,
                      "justifyContent": "center",
                      "marginBottom": 20,
                      "marginLeft": 0,
                      "marginRight": 0,
                      "opacity": 1,
                      "width": 200,
                    }
                  }
                  testID={undefined}>
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                      Array [
                        Object {
                          "fontSize": 46,
                        },
                        Object {
                          "color": "#4DBA75",
                        },
                      ]
                    }>
                    Achei!
                  </Text>
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                      Array [
                        Object {
                          "fontSize": 16,
                        },
                        Object {
                          "color": "#4DBA75",
                        },
                      ]
                    }>
                    Mapear Alimento
                  </Text>
                </View>
              </View>
            </View>
            <View />
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
